python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -856,6 +868,14 @@ def _unpack_object(val: dict, whitelist_map: WhitelistMap, context: UnpackContex
         items = cast(List[JsonSerializableValue], val["__frozenset__"])
         return frozenset(items)
 
+    if "__dict_keyed_by_asset_key_" in val:
+        from dagster._core.events import AssetKey
+
+        return {
+            AssetKey.from_user_string(key): _unpack_value(value, whitelist_map, context)
```

Comment:
`to_user_string` / `from_user_string` doesn't correctly roundtrip multipart asset keys that contain `/` in them 
```
>>> from dagster import AssetKey
>>> a = AssetKey(['this/sucks', 'a////bit'])
>>> a
AssetKey(['this/sucks', 'a////bit'])
>>> a.to_user_string()
'this/sucks/a////bit'
>>> b = AssetKe.from_user_string(a.to_user_string())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'AssetKe' is not defined. Did you mean: 'AssetKey'?
>>> b = AssetKey.from_user_string(a.to_user_string())
>>> b
AssetKey(['this', 'sucks', 'a', '', '', '', 'bit'])
>>> a == b
False
```

Reply:
ahh this is a good catch. Initially I picked `to_user_string` because I thought it would likely be more performant than using JSON, but probably a good idea to instead call `asset_key.to_string()` instead